mixer/pkg/runtime/dispatcher/dispatcher_test.go

```diff
@@ -801,6 +801,41 @@ ident                         : dest.istio-system
 [tcheck] DispatchCheck <= (PANIC)
 `,
 	},
+
+	{
+		name: "CheckElidedRule",
+		config: []string{
+			data.HandlerACheckOutput1,
+			data.InstanceCheckOutput1,
+			data.RuleCheckNoActionsOrHeaderOps,
+		},
+		variety: tpb.TEMPLATE_VARIETY_CHECK_WITH_OUTPUT,
+		expectedCheckResult: adapter.CheckResult{
+			ValidDuration: 60000000000,
+			ValidUseCount: 10000,
+		},
+		log: ``,
+	},
+
+	{
+		name: "CheckOnlyHeaderOperationRule",
+		config: []string{
+			data.HandlerACheckOutput1,
+			data.InstanceCheckOutput1,
+			data.RuleCheckHeaderOpWithNoActions,
+		},
+		variety: tpb.TEMPLATE_VARIETY_CHECK_WITH_OUTPUT,
+		expectedCheckResult: adapter.CheckResult{
```

Comment:
There should be some valid duration set in this case. I think it's stemming from the defaulting logic here https://github.com/istio/istio/blob/master/mixer/pkg/runtime/dispatcher/dispatcher.go#L135. Maybe change it so that it defaults if there is a route directive.

Reply:
Yes, I think I wouldn't touch this function. "Default" is a confusing term. How about change the use of it at the dispatcher? I think the dispatcher needs to populate validity constraints if they are not set, instead of checking against "default".

@geeknoid @douglas-reid @mandarjog thoughts?

Reply:
An alternative could be to move the defaulting out of dispatcher.Check() and into the session.dispatch() function. We could then populate it with default when destinations.Entries() is empty.